Source/OxyPlot/Axes/Rendering/HorizontalAndVerticalAxisRenderer.cs

```diff
@@ -322,13 +332,35 @@ protected virtual void RenderMajorItems(Axis axis, double axisPosition, double t
                 {
                     if (isHorizontal)
                     {
-                        majorSegments.Add(new ScreenPoint(transformedValue, plotAreaTop));
-                        majorSegments.Add(new ScreenPoint(transformedValue, plotAreaBottom));
+                        if (!cropGridlines)
```

Comment:
it looks like this code block is repeated 6 times. Could it be a good candidate for a method or anonymous method?